client/parser/gen.go

```diff
@@ -0,0 +1,391 @@
+					case model.Flat:
+						acc = append(acc, ASTNode{Type: FlatNode})
+					case model.Natural:
+						acc = append(acc, ASTNode{Type: NaturalNode})
+					case model.Sharp:
+						acc = append(acc, ASTNode{Type: SharpNode})
+					}
+				}
+				laa.Children = append(laa.Children, ASTNode{
+					Type:     NoteAccidentalsNode,
+					Children: acc,
+				})
+			}
+
+			note.Children = append(note.Children, laa)
+
+		default:
+			// model.MidiNoteNumber - never parsed from Alda code, only used:
+			// 	1. with model.LispPitch (unnecessary to handle here)
+			// 	2. by MusicXML importer (removed during optimization)
+			return ASTNode{}, fmt.Errorf(
+				"unexpected PitchIdentifier type during AST gen: %#v", pitch,
+			)
+
+		}
+
+		note, err := appendDuration(note, update.Duration)
+		if err != nil {
+			return ASTNode{}, err
+		}
+
+		if update.Slurred {
+			note.Children = append(note.Children, ASTNode{Type: TieNode})
+		}
+
+		return note, nil
+
+	case model.OnRepetitions:
+		node, err := mapIsolatedUpdate(update.Event)
+		if err != nil {
+			return ASTNode{}, err
+		}
+
+		repetitions := ASTNode{Type: RepetitionsNode}
+		for _, rep := range update.Repetitions {
+			repetitions.Children = append(repetitions.Children, ASTNode{
+				Type: RepetitionRangeNode,
+				Children: []ASTNode{
+					{Type: FirstRepetitionNode, Literal: rep.First},
+					{Type: LastRepetitionNode, Literal: rep.Last},
+				},
+			})
+		}
+
+		return ASTNode{
+			Type:     OnRepetitionsNode,
+			Children: []ASTNode{node, repetitions},
+		}, nil
+
+	case model.Repeat:
+		node, err := mapIsolatedUpdate(update.Event)
+		if err != nil {
+			return ASTNode{}, err
+		}
+		return ASTNode{Type: RepeatNode, Children: []ASTNode{
+			node,
+			{Type: TimesNode, Literal: update.Times},
+		}}, nil
+
+	case model.Rest:
+		return appendDuration(ASTNode{Type: RestNode}, update.Duration)
+
+	case model.VariableDefinition:
+		children, err := mapUpdatesWithoutParts(update.Events)
+		if err != nil {
+			return ASTNode{}, err
+		}
+		return ASTNode{Type: VariableDefinitionNode, Children: []ASTNode{
+			{Type: VariableNameNode, Literal: update.VariableName},
+			{Type: EventSequenceNode, Children: children},
+		}}, nil
+
+	case model.VariableReference:
+		return ASTNode{
+			Type:    VariableReferenceNode,
+			Literal: update.VariableName,
+		}, nil
+
+	default:
+		// PartDeclaration, VoiceMarker, VoiceGroupEndMarker - handled upstream
+		// GlobalAttributeUpdate - only generated within Lisp
+		// LispNil - should never exist here
+		return ASTNode{}, fmt.Errorf(
+			"unexpected ScoreUpdate type during AST gen: %#v", update,
+		)
+
+	}
+}
+
+// mapUpdatesWithoutParts maps a group of updates to a group of ASTNode's.
+// Parts are ignored and handled upstream in GenerateASTFromScoreUpdates.
+// Voices are handled by identifying model.VoiceMarker then generating
+// subsequent nodes within an overarching VoiceGroupNode.
+func mapUpdatesWithoutParts(updates []model.ScoreUpdate) ([]ASTNode, error) {
```

Comment:
note for self, makes more sense to call this `mapUpdatesWithinParts` or something like that

Reply:
Refactored to follow parser naming format with "topLevel" and "innerEvents"